### ISSUE TITLE
Document Fallback Behaviour and Functional Requirements

### ISSUE THREAD

**Issue by billgertz** [home-assistant/home-assistant.github.io#19511]
### Feedback

In the recently closed [PR#56](https://github.com/home-assistant/plugin-dns/pull/56) [pvizeli](https://github.com/pvizeli) wrote that the change suggested would "not solve the main issue which we solved with this."

Fine.

If we have our hearts set on retaining this feature with no way to opt out then please document the fallback behaviour. HA users need to be aware that DoH requests to CloudFlare should be expected. Currently, this is an undocumented feature that is causing issues with deployments to small system (for example older RPi platforms, see [this comment](https://github.com/home-assistant/plugin-dns/pull/56#issuecomment-928967969)).

You would also need to point out the need to allow firewall rules are needed to allow the HA deployment to reach the CloudFlare servers on:

- from source (HA Deployment IP) to destination 1.1.1.1 protocol tcp destination port 853
- from source (HA Deployment IP) to destination 1.0.0.1 protocol tcp destination port 853

The behaviour is currently undocumented and unexpected.

### URL

https://www.home-assistant.io/integrations/dnsip/

### Version

2021.9.7

### Additional information

I'd be happy to write this up, please let me know.

**Comment by frenck** [comment]
This is not related to the integration where this feedback was provided on, nor do I have clue on where this would fit.

This applies to managed systems, if you want to manage your own system we provide other installation methods that might be a better fit for you (for example, running Home Assistant Core directly or using the Home Assistant Container).

Nevertheless, feel free to provide a PR with an edit suggestion.

I'm closing the issue on this end, as it is not related to the integration provided feedback on nor it is a subject or documentation covers or should cover (IMHO).

**Comment by billgertz** [comment]
I'm using Home Assistant Container and am seeing this behaviour. So just submit a PR here or is there a document repository where I can suggest and issue a PR?

**Comment by frenck** [comment]
> I'm using Home Assistant Container and am seeing this behaviour

You are not, as the Home Assistant DNS plugin isn't used by the Home Assistant Container installation method. So you are either running something else, or not experiencing the referenced issue.

**Comment by billgertz** [comment]
Yes I am and yes it is a docker install, although it was upgraded from a much older version. I'll try a fresh install on Debian 10 and import the config from the old install. I'll rattle this cage if I still have the issue.

**Comment by frenck** [comment]
> Yes I am and yes it is a docker install.

Again the Docker install is not using the DNS plugin, as that is only used by the Home Assistant Supervised and Home Assistant Operating System installation types. The Home Assistant Core and Home Assistant Container installation types do not use these plugins, as those latter two installation types don't use a Supervisor.

For more information of installation types and their provided functionality, please consult or documentation.

**Comment by frenck** [comment]
> I'll rattle this cage if I still have the issue.

Well this cage would be the wrong one. See my comments above.

**Comment by billgertz** [comment]
Thanks for listening. :-/

**Comment by fenichelar** [comment]
@frenck 

> This applies to managed systems, if you want to manage your own system we provide other installation methods that might be a better fit for you (for example, running Home Assistant Core directly or using the Home Assistant Container).

To clarify, this isn't just a problem for people that want to manage HA themselves. If the DNS plugin cannot communicate with Cloudflare for any reason, it can have major performance implications for HA. https://github.com/home-assistant/plugin-dns/pull/56#issuecomment-928967969 describes what happens. I can confirm this to be accurate, I see the same behavior.

**Comment by frenck** [comment]
> To clarify, this isn't just a problem for people that want to manage HA themselves. If the DNS plugin cannot communicate with Cloudflare for any reason, it can have major performance implications for HA. home-assistant/plugin-dns#56 (comment) describes what happens. I can confirm this to be accurate, I see the same behavior.

Considering this is documentation feedback: That is not a documentation concern.

**Comment by fenichelar** [comment]
> Considering this is documentation feedback: That is not a documentation concern.

Access to Cloudflare's DNS servers is a requirement for HA to function. I didn't know this was required so I blocked TCP port 853. HA then started operating very slowly. It wasn't until I investigated HA code that I found this behavior. This requirement should probably be documented so that people don't block TCP port 853 and find out the hard way.
